autoencoder.py

```diff
@@ -0,0 +1,46 @@
+from torch import nn
+
+"""
+Autoencoder
+
+This model is currently used to reduce the dimensionality of the image vectors produced by the ResNet50 model.
+Thhis model has, in practice, proven itself to be more effective in our image inference tasks
+than various ViT models (primarily google/vit-base-patch16-224).
+
+However, vectors indexed within the PGVector database are limited to 2000 dimensions.
+Therefore, this model has been trained on our parsed image dataset to reduce the
+dimensionality of the vectors generated by the ResNet50 model from 2048 to 2000.
+"""
+
+VECTORIZER_OUTPUT_DIM = 2048
+TARGET_VECTOR_DIM = 2000
```

Comment:
Can you sanity check that we can build 2000 dimension hnsw indices w/pgvector? Just worried of the unlikely scenario that "2000 dimension limit" means <2000 in this case.

Reply:
Good catch I will test

Reply:
Can confirm that indexing works; adding to this pr https://github.com/ConductorAI/masc/pull/136